src/main/java/org/broadinstitute/hellbender/engine/SlidingWindowWalker.java

```diff
@@ -0,0 +1,116 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMSequenceRecord;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import java.util.ArrayList;
+
+/**
+ * An SlidingWindowWalker is a tool that processes a single window over the genome (constructed with a window-size and window-step)
+ * at a time, with the ability to query optional overlapping sources of reads, reference data, and/or variants/features.
+ *
+ * SlidingWindow authors must implement the apply() method to process each position, and may optionally implement
+ * onTraversalStart() and/or onTraversalDone().
+ *
+ * @author Daniel Gómez-Sánchez (magicDGS)
+ */
+public abstract class SlidingWindowWalker extends GATKTool {
+
+	@Argument(fullName = "windowSize", shortName = "windowSize", doc = "Window size for iterate over the variants", common = false, optional = false)
```

Comment:
Also, since `SlidingWindowWalker` is not variant-specific, the docs for these arguments should probably not mention variants.